Duplicati/Library/Backend/Telegram/TelegramBackend.cs

```diff
@@ -0,0 +1,475 @@
+                    return result;
+                },
+                nameof(List));
+        }
+
+        public Task PutAsync(string remotename, string filename, CancellationToken cancelToken)
+        {
+            return SafeExecute(() =>
+                {
+                    var channel = GetChannel();
+                    using (var fs = File.OpenRead(filename))
+                    {
+                        var fsReader = new StreamReader(fs);
+
+                        cancelToken.ThrowIfCancellationRequested();
+                        EnsureConnected();
+                        var file = m_telegramClient.UploadFile(remotename, fsReader, cancelToken).GetAwaiter().GetResult();
+
+                        cancelToken.ThrowIfCancellationRequested();
+                        var inputPeerChannel = new TLInputPeerChannel {ChannelId = channel.Id, AccessHash = (long)channel.AccessHash};
+                        var fileNameAttribute = new TLDocumentAttributeFilename
+                        {
+                            FileName = remotename
+                        };
+
+                        EnsureConnected();
+                        m_telegramClient.SendUploadedDocument(inputPeerChannel, file, remotename, "application/zip", new TLVector<TLAbsDocumentAttribute> {fileNameAttribute}, cancelToken).GetAwaiter().GetResult();
+                        fs.Close();
+                    }
+
+                    return Task.CompletedTask;
+                },
+                $"{nameof(PutAsync)}({remotename})");
+        }
+
+        public void Get(string remotename, string filename)
+        {
+            SafeExecute(() =>
+                {
+                    var fileInfo = ListChannelFileInfos().First(fi => fi.Name == remotename);
+                    var fileLocation = fileInfo.ToFileLocation();
+
+                    var limit = MEBIBYTE_IN_BYTES;
+                    var currentOffset = 0;
+
+                    using (var fs = new FileStream(filename, FileMode.Create, FileAccess.Write))
+                    {
+                        while (currentOffset < fileInfo.Size)
+                        {
+                            try
+                            {
+                                EnsureConnected();
+                                var file = m_telegramClient.GetFile(fileLocation, limit, currentOffset).GetAwaiter().GetResult();
+                                fs.Write(file.Bytes, 0, file.Bytes.Length);
+                                currentOffset += file.Bytes.Length;
+                            }
+                            catch (InvalidOperationException e)
+                            {
+                                if (e.Message.Contains("Couldn't read the packet length") == false)
+                                {
+                                    throw;
+                                }
+                            }
+                        }
+
+                        fs.Close();
```

Comment:
I don't think it's necessary to explicitly call `Close`.  The call to `Dispose` should take care of it.

Reply:
Fixed, thanks